app/helpers/shop_helper.rb

```diff
@@ -20,12 +20,24 @@ def user_is_related_to_distributor?
     )
   end
 
-  def shop_tabs
+  def tabs_with_groups
     [
       { name: 'about', title: t(:shopping_tabs_about, distributor: current_distributor.name), cols: 6 },
       { name: 'producers', title: t(:label_producers), cols: 2 },
       { name: 'contact', title: t(:shopping_tabs_contact), cols: 2 },
-      { name: 'groups', title: t(:label_groups), cols: 2 },
+      { name: 'groups', title: t(:label_groups), cols: 2 }
+    ]
+  end
+
+  def tabs_without_groups
+    [
+      { name: 'about', title: t(:shopping_tabs_about, distributor: current_distributor.name), cols: 4 },
+      { name: 'producers', title: t(:label_producers), cols: 4 },
+      { name: 'contact', title: t(:shopping_tabs_contact), cols: 4 }
     ]
   end
+
+  def shop_tabs
+    current_distributor.groups.present? ? tabs_with_groups : tabs_without_groups
+  end
```

Comment:
I think there's some unnecessary copy paste here. what do you think about something like this?

```ruby
def base_shop_tabs(column_sizes)
   [
      { name: 'about', title: t(:shopping_tabs_about, distributor: current_distributor.name), cols: column_sizes[0] },
      { name: 'producers', title: t(:label_producers), cols: column_sizes[1] },
      { name: 'contact', title: t(:shopping_tabs_contact), cols: column_sizes[2] }
   ]
end

def tabs_with_groups
    tabs = base_shop_tabs([6, 2, 2])
    tabs << { name: 'groups', title: t(:label_groups), cols: 2 }
end

def tabs_without_groups
   base_shop_tabs([4, 4, 4])
end


def shop_tabs
    current_distributor.groups.present? ? tabs_with_groups : tabs_without_groups
end
```

Reply:
Great idea! Thanks